contracts/SimpleVaultMigrator.sol

```diff
@@ -0,0 +1,264 @@
+// SPDX-License-Identifier: AGPL-3.0
+
+pragma solidity ^0.8.15;
+
+import "@newopenzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+interface IVaultAPI is IERC20 {
+    function deposit(uint256 _amount, address recipient)
+        external
+        returns (uint256 shares);
+
+    function withdraw(uint256 _shares) external;
+
+    function token() external view returns (address);
+
+    function permit(
+        address owner,
+        address spender,
+        uint256 value,
+        uint256 deadline,
+        bytes calldata signature
+    ) external returns (bool);
+}
+
+interface IRegistry {
+    function latestVault(address token) external view returns (address);
+}
+
+contract SimpleVaultMigrator {
+    using SafeERC20 for IERC20;
+    using SafeERC20 for IVaultAPI;
+
+    /* ========== STATE VARIABLES ========== */
+
+    /// @notice Governance can update the registry and sweep stuck tokens.
+    address public governance;
+
+    /// @notice New address must be set by current gov and then accepted to transfer power.
+    address public pendingGovernance;
+
+    /// @notice Vault registry to pull info about yearn vaults. This will vary based on network.
+    address public registry;
+
+    /* ========== CONSTRUCTOR ========== */
+
+    constructor(address _governance, address _registry) {
+        require(_governance != address(0), "Governance cannot be 0");
+        require(_registry != address(0), "Registry cannot be 0");
+        governance = _governance;
+
+        registry = _registry;
+    }
+
+    /* ========== EVENTS ========== */
+    event NewGovernance(address indexed governance);
+
+    event NewRegistry(address indexed registry);
+
+    event SuccessfulMigration(
+        address indexed user,
+        address indexed vaultFrom,
+        address indexed vaultTo,
+        uint256 migratedAmount
+    );
+
+    /* ========== MODIFIERS ========== */
+    modifier onlyGovernance {
+        require(msg.sender == governance, "Sender must be governance");
+        _;
+    }
+
+    modifier onlyPendingGovernance {
+        require(
+            msg.sender == pendingGovernance,
+            "Sender must be pending governance"
+        );
+        _;
+    }
+
+    modifier checkVaults(address vaultFrom, address vaultTo) {
```

Comment:
Is it possible to verify that both vaults are legit/valid checking in the registry?

Reply:
well, this is checking that the vault we are migrating to is a legit vault and is the newest version for that token, so I think that's all we should need—we don't really care if someone is migrating a random fake vault or whatever to the newest endorsed version